randovania/games/fusion/exporter/patch_data_factory.py

```diff
@@ -216,6 +218,65 @@ def _create_nav_text(self) -> dict:
             }
         return nav_text_json
 
+    def _credits_elements(self) -> dict:
+        elements = {}
+        for player_index, patches in self.description.all_patches.items():
+            for pickup_index, target in patches.pickup_assignment.items():
+                if target.player != self.players_config.player_index:
+                    continue
+                region_list = default_database.game_description_for(patches.configuration.game).region_list
+                pickup_category = target.pickup.pickup_category
+                if pickup_category.hinted_as_major or pickup_category.is_key:
+                    player_name = None
+                    if self.players_config.is_multiworld:
+                        player_name = self.players_config.player_names[player_index]
+                    if target.pickup.name not in elements:
+                        elements[target.pickup.name] = [
+                            {
+                                "World": player_name,
+                                "Region": region_list.region_name_from_node(
+                                    region_list.node_from_pickup_index(pickup_index), True
+                                ),
+                                "Area": region_list.nodes_to_area(
+                                    region_list.node_from_pickup_index(pickup_index)
+                                ).name,
+                            }
+                        ]
+                    else:
+                        elements[target.pickup.name].append(
```

Comment:
As was mentioned in VC, please get rid of the duplication here. Your choice if you want to use a defaultdict or stuff this into a variable